### ISSUE TITLE
IP Anonymization in Google Analytics

### ISSUE THREAD

**Issue by richpomfret** [FreeUKGen/FreeCENMigration#475]
Not nessary for 25th - as we updating the cookies to include this just in case. However, as part of GDPR, we should implement IP Anonymization in Google Analytics. This will remove any identifying user info and enable us to include analytics cookies as part of our essential cookies.

**Comment by PatReynolds** [comment]
Needed for users to be able to select 'no' to sending data to advertisers (but still getting non-personal adverts)

**Comment by smrr723** [comment]
@PatReynolds / @richpomfret  easy to implement, just need clarification on the below.

Currently, the analytics cookie is either stored, or it isn't, depending on the choice made. Are we proposing to just make their IP anonymous rather than deleting their analytics cookies, if they turn off the analytics option?  If so, will probably need to include text changes on the cookie page to explain we are doing this?

**Comment by PatReynolds** [comment]
I am not sure (@richpomfret can you confirm?) Current text:
We use Google analytics in order to help us understand how the site is used and to help guide improvements.
The text for the cookie page would then need to add: 'if you choose not to share personal data Analytics will anonymizes your address'.
and then in the analytics section:
We use Google analytics in order to help us understand how the site is used and to help guide improvements. Such data might identify the browser you use, device used and whether you are a returning user or not.
would need changing to 
Such data might identify the browser you use, device used and, if you have given  permission to record your IP address, whether you are a returning user or not.

**Comment by smrr723** [comment]
Have tested this, first with plain analytics code and then with tag manager as a trial run.  Figured tag manager out and it's easy enough to run - so would like to add it in for all related stories.

On further reading, I would say we should just default to having everything for Analytics anonymized.  It makes us GDPR compliant, means we don't have to bother giving the user a choice for analytics, and the only actual difference to our data is slightly less accurate city level tracking (can see a users city but street/neighborhood area won't be as accurate), which for us and most websites is a negligible loss.

**Comment by smrr723** [comment]
Since discussing with @richpomfret yesterday - the solution above which was to allow Analytics cookies whilst Anonymising the IP isn’t compatible with our strict GDPR strategy, due to some Ad features we have turned on (though not entirely sure this is correct).

@richpomfret - I can’t see a middle ground where we can choose not to set the “ad related part” of analytics cookies.  We either have Analytics cookies on, or off (currently off by default).

More importantly, all articles I read so far have failed to mention a crucial point which I’d love to have found out earlier - I’ve discovered this week that traffic figures are in-fact affected simply by using the opt-in cookie strategy.  The reason for this is because without a cookie being set to track our users, Google cannot tell the difference between a new user, and a returning one. Having asked around - it looks like our cookie strategy is responsible for the traffic increase, and not a code error like originally thought.

If we continue giving users the choice only to opt-in, our analytics data will be impacted as we’ve seen to date so far.  So, going forward - our choices are to either leave things as they are and accept the data impact, or we set analytics cookies on as default (whilst Anon IP) - I can’t see any other alternatives but open to ideas if anyone comes across something.
